python/tvm/contrib/target/onnx.py

```diff
@@ -510,6 +510,99 @@ def convert(cls, node_entry, model_container, node_dict):
                                            axis=axis)
         model_container.add_nodes([slice_node])
 
+class LayoutTransform(OpConverter):
+    """ Operator converter for Layouttransform
+    """
+
+    @classmethod
+    def convert_attributes(cls, attrs):
+        src_layout = attrs.get_str("src_layout")
+        dst_layout = attrs.get_str("dst_layout")
+
+        perm = [src_layout.index(c) for c in dst_layout]
+        return {'perm': tuple(perm)}
+
+    @classmethod
+    def convert(cls, node_entry, model_container, node_dict):
+        attrs = cls.convert_attributes(node_entry['relay_node'].attrs)
+        onnx_node = onnx.helper.make_node("Transpose",
+                                          node_entry['input_names'],
+                                          node_entry['output_names'],
+                                          **attrs)
+        model_container.add_nodes([onnx_node])
+
+class Clip(OpConverter):
+    """ Operator converter for Clip.
+    """
+
+    @classmethod
+    def convert_attributes(cls, attrs):
+        return {
+            'min': attrs.a_min,
+            'max': attrs.a_max
+        }
+
+    @classmethod
+    def convert(cls, node_entry, model_container, node_dict):
+        attrs = cls.convert_attributes(node_entry['relay_node'].attrs)
+
+        name = node_entry['name']
+
+        def _add_input(val, input_name):
+            val_arr = numpy.asarray(val).astype(numpy.float32)
+            input_name = '{}_{}'.format(name, input_name)
+            add_input(val_arr, input_name, model_container)
+            return input_name
+
+        input_names = []
+        input_names.append(_add_input(attrs['min'], 'min'))
+        input_names.append(_add_input(attrs['max'], 'max'))
+
+        input_names = [node_entry['input_names'][0]] + input_names
+
+        node = onnx.helper.make_node(cls.__name__, input_names, node_entry['output_names'])
+        model_container.add_nodes([node])
+
+class Expand(OpConverter):
+    """ Operator converter for Expand_dims.
+    """
+
+    @classmethod
+    def convert_attributes(cls, attrs):
+        return {
+            'axis': attrs.axis,
+            'num_newaxis': attrs.num_newaxis
+        }
+
+    @classmethod
+    def convert(cls, node_entry, model_container, node_dict):
+        attrs = cls.convert_attributes(node_entry['relay_node'].attrs)
+
+        name = node_entry['name']
+
+        def _add_input(val, input_name):
+            val_arr = numpy.asarray(val).astype(numpy.int64)
+            input_name = '{}_{}'.format(name, input_name)
+            add_input(val_arr, input_name, model_container)
+            return input_name
```

Comment:
It seems this function is now repeated quite a few times in `OpConverter` subclasses. Wouldn't be interesting to make it a common implementation, provided by `OpConverter` itself?

Reply:
While I agree with @leandron that this helper function appears many times in different ops, it seems improper to put `_add_input` to `OpConverter` because 1) this is not required by all subclasses; 2)we already have a static method `add_input` and `_add_input` is more like a helper for each op.

I'd suggest enhance `add_input` to support 1) numpy array data and 2) input name prefix. In this way, `add_input` itself should be sufficient.

Reply:
Yep. Agree with @comaniac

Reply:
Agree, will address comments in a new commit